File: server/models/user.model.js
const mongoose = require('mongoose')

const userSchema = new mongoose.Schema({
  email: { type: String, required: true },
  password: { type: String, required: true },
  displayName: { type: String, required: true },
  birthday: { type: String, required: true },
  status: { type: String, default: 'Online' },
  username: { type: String, required: true, unique: true },
  bio: { type: String, required: false },
  profilePic: {
    type: String,
    default:
      'https://archive.org/download/discordprofilepictures/discordgrey.png',
  },
  createdAt: { type: Date, default: Date.now },
})

const User = mongoose.model('User', userSchema)

module.exports = User

File: client/lib/model/user.model.dart
class User {
  final String id;
  final String email;
  final String displayName;
  final String username;
  final String birthday;
  final String createdAt;
  final String profilePic;
  final String status;

  User(
      {required this.id,
      required this.email,
      required this.displayName,
      required this.username,
      required this.birthday,
      required this.createdAt,
      required this.profilePic,
      required this.status});
}
